### PR TITLE
MOTS: 351 Added validation to create Community and Facility. 

### PR DESCRIPTION
**Pull request by pstudzinski**
Added validation to create Community and Facility.
Refactor in LocationController.
Adjusted UI to new DTOs of community and facility.
https://applab.atlassian.net/browse/MOTS-351

### CODE REVIEW

File: src/main/java/org/motechproject/mots/dto/CommunityCreationDto.java [hunk 1]
```diff
@@ -15,19 +17,13 @@
...
-  @Setter
-  @Getter
-  private String chiefdomId;
-
-  @Setter
-  @Getter
-  @NotBlank(message = ValidationMessages.EMPTY_FACILITY_TYPE)
+  @FacilityExistence
+  @Uuid
+  @NotEmpty(message = ValidationMessages.EMPTY)
```
Comment (kkaczmarczyk) [inline src/main/java/org/motechproject/mots/dto/CommunityCreationDto.java]: Let's bring back full validation message. The generic validation will be added later on

Comment (pstudzinski) [reply src/main/java/org/motechproject/mots/dto/CommunityCreationDto.java]: Corrected and added missing validations.

File: src/main/java/org/motechproject/mots/constants/ValidationMessages.java [hunk 2]
```diff
@@ -41,18 +42,24 @@
...
   public static final String NOT_EXISTING_DISTRICT = "District doesn't exist";
+  public static final String NOT_EXISTING_CHIEFDOM = "Chiefdom doesn't exist";
   public static final String NOT_EXISTING_MODULES_WITH_ID = "Module with id: %s doesn't exist";
 
   public static final String END_DATE_AFTER_START_DATE = "End date must be after start date.";
 
   public static final String NOT_UNIQUE_USERNAME = "The User with username '%s' already exists";
+  public static final String NOT_UNIQUE_FACILITY_ID = "The Facility with this ID already exists";
+  public static final String NOT_UNIQUE_FACILITY =
+      "The Facility with name: '%s' and chiefdom: '%s' already exists";
```
Comment (kkaczmarczyk) [inline src/main/java/org/motechproject/mots/constants/ValidationMessages.java]: and -> within

File: src/main/java/org/motechproject/mots/constants/ValidationMessages.java [hunk 3]
```diff
@@ -41,18 +42,24 @@
...
+  public static final String NOT_UNIQUE_COMMUNITY =
+      "The Community with name: '%s' and facility: '%s' already exists";
```
Comment (kkaczmarczyk) [inline src/main/java/org/motechproject/mots/constants/ValidationMessages.java]: and -> within

File: src/main/java/org/motechproject/mots/dto/CommunityCreationDto.java (hunk 1 above)
Comment (kkaczmarczyk) [reply src/main/java/org/motechproject/mots/dto/CommunityCreationDto.java]: it's still using ValidationMessages.EMPTY as opposed to full message. is that on purpose?

**Comment by pstudzinski** [comment]
@kkaczmarczyk  I fixed messages after review.

**Pull request merged**